python/paddle/distributed/launch/context/device.py

```diff
@@ -112,11 +121,28 @@ def parse_device(self):
     @classmethod
     def detect_device(self):
         import paddle.fluid as fluid
```

Comment:
import paddle is slow and expensive, this function may not be used in most cases.

Reply:
`The only good reason to import at function level is to avoid circular dependencies.`
https://stackoverflow.com/questions/37067414/python-import-multiple-times.
https://stackoverflow.com/questions/20346189/which-is-a-better-practice-global-import-or-local-import.

Reply:
In most times, this function will be called.
<img width="728" alt="image" src="https://user-images.githubusercontent.com/10721757/182519397-fd1da770-e69d-4afc-a0bc-5a6c46218bba.png">

Reply:
This is a trade-off, in this context, import inner function is better and no risk of circular dependency.